lib/ember_cli/shell.rb

```diff
@@ -50,6 +52,23 @@ def test
     attr_accessor :pid
     attr_reader :ember, :env, :options, :paths
 
+    def invalid_ember_dependencies?
+      !run("#{paths.ember} version")
+    rescue DependencyError
```

Comment:
In a similar vein of `Runner#run` and `Runner#run!`, maybe the best way forward is to split `PathSet#ember` into `PathSet#ember` and `PathSet#ember!`, where the latter explodes?